File: week2/Eren-Taskiran/121-best-time-to-buy-and-sell-stock.c
int maxProfit(int* prices, int pricesSize) {
    int min=prices[0];
    int potentialProfit=0;
    int maxProfit=0;
    int i;
    for(i = 0; i<pricesSize; i++){
        if(min>prices[i]) min=prices[i];
        potentialProfit=prices[i]-min;
        if(potentialProfit>maxProfit) maxProfit=potentialProfit;
    }
    return maxProfit;
}

File: week2/furkanunsalan/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self,prices):
        buy = 0
        sell = 1
        max_profit = 0
        while sell < len(prices):
            currentProfit = prices[sell] - prices[buy]
            if prices[buy] < prices[sell]:
                max_profit = max(currentProfit,max_profit)
            else:
                buy = sell
            sell += 1
        return max_profit